Tests/scripts/jira_issues.py

```diff
@@ -88,3 +96,33 @@ def jira_server_information(jira_server: JIRA):
     logging.info("Jira server information:")
     for key, value in jira_server_info.items():
         logging.info(f"\t{key}: {value}")
+
+
+def jira_search_all_by_query(jira_server: JIRA,
+                             jql_query: str,
+                             max_results_per_request: int = 100) -> dict[str, list[Issue]]:
+
+    # Initialize pagination parameters
+    start_at = 0
+    total_issues = float('inf')  # Set an initial value to enter the loop.
+    issues: dict[str, list[Issue]] = {}
+    while start_at < total_issues:
```

Comment:
Why not check the length of `issues`?
And do `break`, instead of the 2 variables `start_at` and `total_issues`.

Reply:
will change it.